scanners/zap/zap.py

```diff
@@ -661,8 +670,6 @@ def _validate_rule_boolean_values(rule):
             raise ValueError("replacer must have a rule at least")
 
         delete_all_rules = self.my_conf("replacer.parameters.deleteAllRules", default=True)
```

Comment:
Probably out of scope for this PR, but with this change can we replace lines like this with something like below?

```suggestion
        delete_all_rules = cfg.replacer.parameters.deleteAllRules
```

(lines like this might even be completely unnecessary if we can just replace occurrences of `delete_all_rules` with `cfg.replacer.parameters.deleteAllRules`)

Reply:
Right, I started working on it, but since `my_conf` was used throughout the codebase, I thought it would be better to handle it in a separate PR to keep the scope of this one limited